### PR TITLE
[#6327] Server does not log stacktrace for config properties that have a default value (4-2-stale)

### PR DESCRIPTION
**Pull request by korydraughn**
Ported from d9a85abce36cdf496b59c0b570509f50e3e6571a

### CODE REVIEW

File: server/re/src/reSysDataObjOpr.cpp [hunk 1]
```diff
@@ -1294,8 +1296,10 @@ msiSetRandomScheme( ruleExecInfo_t *rei ) {
  *
  * \usage See clients/icommands/test/rules/
  *
- * \param[in] xnumProc - a STR_MS_T representing number of processes
+ * \param[in] xnumProc - a STR_MS_T representing number of executors
  *     - this value can be "default" or an integer
+ *     - this value will be clamped to "maximum_number_of_concurrent_rule_engine_server_processes" if defined in server_config.json.
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply server/re/src/reSysDataObjOpr.cpp:1301]: No. The msi will clamp the requested number of executors if it ever exceeds `maximum_number_of_concurrent_rule_engine_server_processes`.

[the remark this answers is not in the document]
Comment (alanking) [reply server/re/src/reSysDataObjOpr.cpp:1301]: Oh, I see. I am not familiar with that term, apparently.

File: server/core/src/irodsReServer.cpp [hunk 2]
```diff
@@ -649,10 +660,11 @@ int main(int, char* _argv[])
                 return bytes;
             }
         }
-        catch (const irods::exception& e) {
-            rodsLog(LOG_WARNING, e.what());
-            rodsLog(LOG_WARNING, "Could not retrieve delay queue byte limit from configuration. "
-                                 "Delay server will use as much memory as necessary.");
+        catch (...) {
+            rodsLog(LOG_WARNING,
+                    "Could not retrieve [%s] from advanced settingsconfiguration. "
```
[the remark this answers is not in the document]
Comment (korydraughn) [reply server/core/src/irodsReServer.cpp]: Fixed.

**Comment by korydraughn** [comment]
Will run this through the testing environment.

File: server/re/src/reSysDataObjOpr.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (trel) [reply server/re/src/reSysDataObjOpr.cpp:1301]: perhaps... "limited" or "capped" rather than "clamped"

[the remark this answers is not in the document]
Comment (korydraughn) [reply server/re/src/reSysDataObjOpr.cpp:1301]: Clamped is a common word used to lock a value to a specific range.

The C++ standard actually contains a function called `std::clamp()`. See the following:
- https://en.cppreference.com/w/cpp/algorithm/clamp
- https://en.wikipedia.org/wiki/Clamping_(graphics)

**Comment by korydraughn** [comment]
All ubuntu github containers are experiencing issues (outside of our control).

Let's ignore them for now.

**Comment by korydraughn** [comment]
All unit tests passed.

Running core tests.

File: server/re/src/reSysDataObjOpr.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (trel) [reply server/re/src/reSysDataObjOpr.cpp:1301]: well, i did not know that.   thanks.

[the remark this answers is not in the document]
Comment (korydraughn) [reply server/re/src/reSysDataObjOpr.cpp:1301]: I'm fine with changing it if we feel one of the other words you presented makes things clearer.

[the remark this answers is not in the document]
Comment (trel) [reply server/re/src/reSysDataObjOpr.cpp:1301]: This is a tension between technical, which is fine as you have pointed to where others have used this language... and writing for everyone and maximum clarity, including non-native english speakers.

I think this is fine as is - it's only in the .cpp file / Doxygen, as opposed to irods_docs.

**Comment by korydraughn** [comment]
All core tests passed.

**Comment by korydraughn** [comment]
#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**